tests/unit/algorithms/action/adapters/mmaction/data/pipelines/test_action_loading.py

```diff
@@ -0,0 +1,100 @@
+"""Unit Test for otx.algorithms.action.adapters.mmaction.data.pipeline.loading.."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+from typing import List
+
+import numpy as np
+import pytest
+
+from otx.algorithms.action.adapters.mmaction.data.cls_dataset import OTXActionClsDataset
+from otx.algorithms.action.adapters.mmaction.data.pipelines.loading import (
+    RawFrameDecode,
+)
+from otx.algorithms.action.configs.classification.x3d.data_pipeline import (
+    train_pipeline,
+)
+from otx.api.entities.annotation import (
+    Annotation,
+    AnnotationSceneEntity,
+    AnnotationSceneKind,
+)
+from otx.api.entities.dataset_item import DatasetItemEntity
+from otx.api.entities.datasets import DatasetEntity
+from otx.api.entities.id import ID
+from otx.api.entities.image import Image
+from otx.api.entities.label import Domain, LabelEntity
+from otx.api.entities.metadata import MetadataItemEntity, VideoMetadata
+from otx.api.entities.scored_label import ScoredLabel
+from otx.api.entities.shapes.rectangle import Rectangle
+from tests.test_suite.e2e_test_system import e2e_pytest_unit
+from tests.unit.algorithms.action.adapters.mmaction.data.test_action_cls_dataset import (
+    MockPipeline,
+)
+
+
+class MockImage(Image):
+    """Mock class for Image entity."""
+
+    @property
+    def numpy(self) -> np.ndarray:
+        """Returns empty numpy array"""
+        return np.ndarray((256, 256))
+
+
+class TestRawFrameDecode:
+    """Test RawFrameDecode class.
+
+    <Steps>
+        1. Create sample OTXActionClsDataset
+        2. Get sample inputs from sample OTXActionClsDataset
+        3. Add "frame_inds", "gt_bboxes", "proposals" attributes to sample inputs
+        4. Check RawFrameDecode transform's results
+            1. Whether transform creates imgs
+            2. Whether transform creates proper img size
+            3. Whether transform modify gt_bboxes and proposals w.r.t img_size
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(self) -> None:
```

Comment:
I think some of codes are duplicated across the test files. how about making common things as a single fixture and use it?